### PR TITLE
Dependency sorting

### PR DESCRIPTION
**Pull request by coolsquid**
Dependencies are declared in the MANIFEST.MF files under the "Dependencies" field. The field contains a list, separated by commas, that specifies each dependency's id, and optionally the minimum version required. Mods are sorted topologically.

Example:
`Dependencies: test1, test2 : 1.0.3, test3 : 2.3.0`

### CODE REVIEW

File: src/main/java/xyz/openmodloader/modloader/ModContainer.java [hunk 1]
```diff
@@ -45,10 +46,12 @@
     private String logoString;
     @SerializedName("Transformer")
     private String transformer;
+    @SerializedName("Dependencies")
+    private String dependencies;
 
     /**
      * Uses a manifest to create a mod container.
-     * Best idea in the entire library.
+     * Worst idea in the entire library.
```
Comment (shadowfacts) [inline src/main/java/xyz/openmodloader/modloader/ModContainer.java]: pls

File: src/main/java/xyz/openmodloader/modloader/ModContainer.java [hunk 2]
```diff
@@ -146,6 +149,19 @@ public String getUpdateURL() {
...
+    public String[] getDependencies() {
+    	if (dependencies == null || dependencies.matches("\\s*"))
+    		return new String[0];
+    	return dependencies.split("\\s*,\\s*");
+    }
+
+    Set<String> getDependencySet() {
+    	Set<String> set = Sets.newHashSet();
+    	for (String s: getDependencies())
+    		set.add(s.split("\\s:\\s")[0]);
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/modloader/ModContainer.java]: pls

File: src/main/java/xyz/openmodloader/modloader/ModContainer.java [hunk 3]
```diff
@@ -86,7 +89,7 @@ public static ModContainer create(Manifest manifest) {
 
     @Strippable(side = Side.CLIENT)
     public ResourceLocation getLogo() {
-        if (this.logo == null && this.logoString != null) {
+        if (this.logo == null) {
```
Comment (shadowfacts) [inline src/main/java/xyz/openmodloader/modloader/ModContainer.java]: What? You're going to cause a crash if a mod author hasn't specified a logo.

File: src/main/java/xyz/openmodloader/modloader/ModContainer.java [hunk 4]
```diff
@@ -146,6 +149,19 @@ public String getUpdateURL() {
     }
 
     public String getTransformer() {
-        return transformer;
```
Comment (shadowfacts) [inline src/main/java/xyz/openmodloader/modloader/ModContainer.java]: These three methods are all using tabs, change them to spaces.

**Comment by iLexiconn** [comment]
Maybe add a `Version` object to the `ModContainer`?

**Pull request merged**